PasswordValidator/Validators/AdvancedValidator.cs

```diff
@@ -1,18 +1,33 @@
 ﻿using PasswordValidator.Enums;
+using PasswordValidator.Rules;
 
 namespace PasswordValidator.Validators
 {
     public class AdvancedValidator : IPasswordValidator
     {
         public ValidatorType ValidatorType => ValidatorType.Advanced;
+        private readonly IEnumerable<IRule> Rules;
```

Comment:
private variables at a class level should be prefixed with _ so this becomes **_rules**